### PR TITLE
Update constraints to torch 2.4 dynamic_shapes API

### PR DESCRIPTION
**Pull request by saienduri**
This commit updates to use dynamic_shapes for dynamic dimensions as the usage of constraints is deprecated in torch 2.4. (The SHARK test checks out the main branch of this repo and will pass once this is merged)

### CODE REVIEW

File: models/turbine_models/custom_models/stateless_llama.py [hunk 1]
```diff
@@ -343,17 +344,19 @@ def run_cached_initialize(
...
+                    "state_arg0_dim1", max=MAX_STEP_SEQ - 1
                 )
+                x_dim = torch.export.Dim("x_dim", max=MAX_STEP_SEQ - 1)
+                dynamic_shapes_forw = {
+                    "arg0_1": {1: x_dim},
+                    "arg1_1": {1: state_arg0_dim1},
+                }
+                for dim_number in range(1, len(state_arg)):
+                    current_dim_dict = {f"arg{dim_number + 1}_1": {1: state_arg0_dim1}}
+                    dynamic_shapes_forw = {**dynamic_shapes_forw, **current_dim_dict}
```
Comment (dan-garvey) [inline models/turbine_models/custom_models/stateless_llama.py:357]: does this need to be inside the for loop?

File: models/turbine_models/custom_models/stateless_llama.py [hunk 2]
```diff
@@ -343,17 +344,19 @@ def run_cached_initialize(
...
+                state_arg0_dim1 = torch.export.Dim(
+                    "state_arg0_dim1", max=MAX_STEP_SEQ - 1
                 )
+                x_dim = torch.export.Dim("x_dim", max=MAX_STEP_SEQ - 1)
+                dynamic_shapes_forw = {
+                    "arg0_1": {1: x_dim},
+                    "arg1_1": {1: state_arg0_dim1},
+                }
+                for dim_number in range(1, len(state_arg)):
+                    current_dim_dict = {f"arg{dim_number + 1}_1": {1: state_arg0_dim1}}
```
Comment (dan-garvey) [inline models/turbine_models/custom_models/stateless_llama.py:356]: really really nit and probably a preference thing, but I think its slightly more readable to make it so dim_number is the actual arg index by adjusting the range arguments.

**Review by dan-garvey — APPROVED** [review]
The builder failures seem unrelated, some minor nits but otherwise looks good

File: models/turbine_models/custom_models/stateless_llama.py (hunk 1 above)
Comment (saienduri) [reply models/turbine_models/custom_models/stateless_llama.py:357]: Yup, we are basically adding another entry to the dictionary for every state arg dim constraint. It's just unpacking both dictionaries and combining them

File: models/turbine_models/custom_models/stateless_llama.py (hunk 2 above)
Comment (saienduri) [reply models/turbine_models/custom_models/stateless_llama.py]: Makes sense, done
